pkg/sql/catalog/replication/reader_catalog_test.go

```diff
@@ -304,52 +304,75 @@ func TestReaderCatalogTSAdvance(t *testing.T) {
 
 	require.NoError(t, advanceTS(now))
 	compareEqual("SELECT * FROM t1")
-	// Validate multiple advances of the timestamp work concurrently with queries.
-	// The tight loop below should relatively easily hit errors if all the timestamps
-	// are not line up on the reader catalog.
-	grp := ctxgroup.WithContext(ctx)
-	require.NoError(t, err)
-	var iterationsDone atomic.Bool
+
 	var newTS hlc.Timestamp
-	grp.GoCtx(func(ctx context.Context) error {
-		defer func() {
-			iterationsDone.Swap(true)
-		}()
-		const NumIterations = 16
-		for iter := 0; iter < NumIterations; iter++ {
-			if _, err := srcRunner.DB.ExecContext(ctx,
-				"INSERT INTO t1(val, j) VALUES('open', $1);",
-				iter); err != nil {
-				return err
-			}
-			// Signal the next timestamp value.
-			newTS = ts.Clock().Now()
-			// Advanced the timestamp next.
-			if err := advanceTS(newTS); err != nil {
-				return err
+
+	for _, useAOST := range []bool{false, true} {
+		errorDetected := false
+		checkAOSTError := func(err error) {
+			if useAOST || err == nil {
+				require.NoError(t, err)
+				return
 			}
+			require.ErrorContains(t, err, "PCR reader timestamp has moved forward, existing descriptor")
```

Comment:
what's the motivation behind this extra testing logic? We still fail the test if !useAOST when `errorDetected` is true.